cmd/resource-topology-exporter/main.go

```diff
@@ -25,13 +23,25 @@ const (
 	ProgramName = "resource-topology-exporter"
 )
 
+type progArgs struct {
+	nrtupdater      *nrtupdater.Args
+	resourcemonitor *resourcemonitor.Args
+	rte             *resourcetopologyexporter.Args
+	version         *bool
+}
+
 func main() {
-	nrtupdaterArgs, resourcemonitorArgs, rteArgs, err := argsParse(os.Args[1:])
+	pArgs, err := parseArgs(os.Args[1:]...)
 	if err != nil {
-		log.Fatalf("failed to parse command line: %v", err)
+		log.Fatalf("failed to parse args: %v", err)
+	}
+
+	if *pArgs.version {
```

Comment:
I don't think you need to deference `pArgs` anyway

Reply:
Right. My C++ nature

Reply:
ok, it's a bit redundant and non-idiomatic, I'd say:
https://golang.org/ref/spec#Selectors

Reply:
Yes as I said, need to check. np I'll change that.

Reply:
hmm I checked and I can't refer to the ```version``` without dereferencing ```pArgs``` first